src/utils/cancelRequests.ts

```diff
@@ -0,0 +1,24 @@
+import axios, { CancelTokenSource, CancelToken } from 'axios';
+
+export class CancelFactory {
+  protected source: CancelTokenSource;
+  private constructor() {
+    this.source = axios.CancelToken.source();
+  }
+
+  public static createSource = (): CancelFactory => {
+    return new CancelFactory();
```

Comment:
With factory patter, you are not creating the object of `CancelFactory` - you should be creating the target object, in this case `CancelTokenSource` or `CancelToken` (depends on the interface you decide to implement). 

In ~~other words~~ code:
```
export class CancelFactory {
  protected static source: cancelTokenSource | null = null;

  public static createSource(): CancelTokenSource {
    if (CancelFactory.cancelTokenSource === null) {
      CancelFactory.cancelTokenSource = axios.CancelToken.source();
    }
    return CancelFactory.cancelTokenSource;
  }
}
```

Note that you don't need the `getToken` and `cancel` methods because they should be called on returned value (source) anyway.